src/library/dao/analysisdao.cpp

```diff
@@ -323,14 +323,26 @@ void AnalysisDao::saveTrackAnalyses(TrackInfoObject* pTrack) {
     ConstWaveformPointer pWaveform = pTrack->getWaveform();
     ConstWaveformPointer pWaveSummary = pTrack->getWaveformSummary();
 
+    TrackId trackId(pTrack->getId());
+
+    // Delete waveform analysis if track was requested to have its waveform cleared.
+    if (pTrack->isClearWaveformRequested()) {
```

Comment:
Sure :) Thanks for your work, Nino! It's a pleasure to review your commits, 
well done.

On 01/11/2016 11:06 PM, Nino MP wrote:
>
> In src/library/dao/analysisdao.cpp 
> <https://github.com/mixxxdj/mixxx/pull/841#discussion_r49386413>:
>
> > @@ -323,14 +323,26 @@ void AnalysisDao::saveTrackAnalyses(TrackInfoObject* pTrack) {
> >      ConstWaveformPointer pWaveform = pTrack->getWaveform();
> >      ConstWaveformPointer pWaveSummary = pTrack->getWaveformSummary();
> >
> > +    TrackId trackId(pTrack->getId());
> > +
> > +    // Delete waveform analysis if track was requested to have its waveform cleared.
> > +    if (pTrack->isClearWaveformRequested()) {
>
> Yes, you are right, but I'm planning to implement automatic clear of 
> waveforms, beats and keys of track being loaded if Mixxx detects that 
> audio of the track has changed, so I need a name that will cover both 
> (automatic and manual) clearing. Do you mind if I do this then?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/mixxxdj/mixxx/pull/841/files#r49386413>.
>